libfwupdplugin/fu-udev-device.c

```diff
@@ -1259,23 +1259,62 @@ fu_udev_device_match_subsystem_devtype(GUdevDevice *udev_device,
 	}
 	return TRUE;
 }
+#endif
 
-static GUdevDevice *
-fu_udev_device_get_parent_with_subsystem_devtype(GUdevDevice *udev_device,
+/**
+ * fu_udev_device_get_parent_with_subsystem_devtype
+ * @self: a #FuUdevDevice
+ * @subsystem: (nullable): the name of a udev subsystem
+ * @devtype: (nullable): the name of a udev devtype, e.g. `usb_interface`
+ * @error: (nullable): optional return location for an error
+ *
+ * Get the device that is a parent of self and has the provided subsystem and device type.
```

Comment:
It keeps going up right?
I think it should be ancestor.

Reply:
Agreed, fixed. Thanks.